python/xorq/vendor/ibis/backends/__init__.py

```diff
@@ -1568,9 +1571,19 @@ def validate_con_name(self, attr, value):
             None,
         )
 
+    def __attrs_post_init__(self):
+        # Sort kwargs_tuple after initialization
+        object.__setattr__(self, "kwargs_tuple", tuple(sorted(self.kwargs_tuple)))
+
+    @property
+    def kwargs_dict(self):
+        return {k: v for k, v in self.kwargs_tuple}
+
     @property
     def hash_name(self):
-        return dask.base.tokenize(self.as_json())
+        pre_hash = json.loads(self.as_json())
+        pre_hash.pop("idx")
```

Comment:
why are you `pop`ing idx?

Reply:
Because the IDX forces uniqueness. I originally wanted to not have hashes differ only on IDX, but this may also solves daniels comment about duplicate profiles.

Reply:
I was thinking the relationships would go:
- Profiles:profile 1:n
- profile:connections 1:m
And that we can instantiate as many connections from a profile as we want.
Because if the connection params are the same thenwhy do we care? Why not let the user manage instances of connections ?